src/main/java/io/fabric8/vertx/maven/plugin/mojos/MojoSpy.java

```diff
@@ -122,12 +122,25 @@ public void mojoStarted(ExecutionEvent executionEvent) {
     public void mojoSucceeded(ExecutionEvent executionEvent) {
         // Unlike mojoStarted, this callback has the lifecycle phase set.
         MojoExecution execution = executionEvent.getMojoExecution();
-        MOJOS.add(execution);
+        addExecutionIfNotContainedAlready(execution);
         if (delegate != null) {
             delegate.mojoSucceeded(executionEvent);
         }
     }
 
+    private void addExecutionIfNotContainedAlready(MojoExecution execution) {
+        String artifact = execution.getArtifactId();
+        String id = execution.getExecutionId();
+        // We must avoid duplicates in the list
+        for (MojoExecution exec : MOJOS) {
+            if (exec.getArtifactId().equals(artifact)  && exec.getExecutionId().equals(id)) {
+                // Duplicate found.
+                return;
+            }
+        }
+        MOJOS.add(execution);
+    }
+
```

Comment:
cant we not use Set  ? so that uniqueness is maintained and we to do explicit check ?